DOCS-3174: Update machine settings (#3930)
viamrobotics/docs: docs/manage/reference/viam-agent/_index.md

<!-- prettier-ignore -->
| Name       | Type | Required? | Description |
| ---------- | ---- | --------- | ----------- |
| `agent` | string | true | The version of Viam agent specified as `"5.6.77"`, `"stable"` or by providing a URL such as `"http://example.com/viam-agent-test-aarch64"` or `"file://home/myuser/viam-agent-test-aarch64"`. Viam agent is semantically versioned and is tested before release. Releases happen infrequently. When set to `"stable"`, `viam-agent` will automatically upgrade when a new version is released. Default: `"stable"`. |
| `viam-server` | string | true | The version of `viam-server` specified as `"5.6.77"`, `"stable"` or by providing a URL such as `"http://example.com/viam-agent-test-aarch64"` or `"file://home/myuser/viam-server-test-aarch64"`. `viam-server` is semantically versioned and is tested before release. When set to `"stable"`, `viam-server` will automatically upgrade when a new stable version is released. Default: `"stable"`. |

{{< alert title="Important" color="note" >}}

> When `viam-agent` updates itself, you must [restart `viam-agent`](/manage/reference/viam-agent/manage-viam-agent/) or reboot in order to use the new version.

Author: ```suggestion
`viam-agent` does not update itself.
You must [restart `viam-agent`](/manage/reference/viam-agent/manage-viam-agent/) or reboot in order to use the new version.
```

You can restart `viam-server` from the machine's part status dropdown to the right of your machine’s name on its page in the Viam app.

{{<imgproc src="configure/machine-part-info.png" resize="500x" declaredimensions=true alt="Restart button on the machine part info dropdown">}}

{{< /alert >}}

For more information on managing `viam-agent` see:

{{< cards >}}
{{% card link="/manage/reference/viam-agent/manage-viam-agent/" %}}
{{< /cards >}}

## `advanced_settings`

<!-- prettier-ignore -->
| Name       | Type | Required? | Description |
| ---------- | ---- | --------- | ----------- |
| `debug` | boolean | false | Sets the log level to debug for any logging from the Viam agent binary. Default: `false`. |
| `disable_network_configuration` | boolean | false | Disables the network and hotspot configuration, as well as the configuration of additional networks. Default: `false`. |
| `disable_system_configuration` | boolean | false | Disables the system configuration. Default: `false`. |
| `disable_viam_server` | boolean | false | Disable `viam-server` remotely. This option is often used by developers working on Viam agent or when manually running `viam-server`. Default: `false`. |

> | `viam_server_start_timeout_minutes` | integer | false | TODO. Default: `10`. |
> | `wait_for_update_check` | boolean | false | If set to `true`, `viam-agent` will wait for a network connection and check for updates before starting `viam-server`. See [Reduce startup time](#reduce-startup-time). Default: `true`. |

Author: ```suggestion
| `viam_server_start_timeout_minutes` | integer | false | Specify a time after which, if `viam-server` hasn't successfully started, Viam agent will kill it and restart. Default: `10`. |
| `wait_for_update_check` | boolean | false | If set to `true`, `viam-agent` will wait for a network connection and check for updates before starting `viam-server`. See [Reduce startup time](#reduce-startup-time). Default: `false`. |
```

| `user_idle_minutes` | integer | false | Amount of time before considering a user (using the captive web portal or provisioning app) idle, and resuming normal behavior. Used to avoid interrupting provisioning mode (for example for network tests/retries) when a user might be busy entering details. Default: `5` (5 minutes). |
| `wifi_power_save` | boolean | false | If set, will explicitly enable or disable power save for all WiFi connections managed by NetworkManager. If not set, the system default applies. Default: `false`. |

For more detailed instructions on what these settings do, see [Provisioning](https://docs.viam.com/manage/fleet/provision/setup/#configure-agent-provisioning).

## `additional_networks`

> For an already-online device, you can configure new WiFi networks in the machine's [`viam-agent` configuration](/manage/reference/viam-agent/#configuration) in the Viam app.
> It's primarily useful for a machine that moves between different networks, so the machine can automatically connect when moved between locations.

Author: ```suggestion
For an already-online device, you can configure new WiFi  or wired networks in the machine's [`viam-agent` configuration](/manage/reference/viam-agent/#configuration) in the Viam app.
It's primarily useful for a machine that moves between different networks, so the machine can automatically connect when moved between locations.
```

<!-- prettier-ignore -->
| Name       | Type | Required? | Description |
| ---------- | ---- | --------- | ----------- |
| `interface` | string | optional | Name of interface, for example: `"wlan0"`, `"eth0"`, `"enp14s0"`. Default: `""`. |
| `ipv4_address` | string | optional | IPv4 address in CIDR format, for example: `"192.168.0.1/24"`. Default: `"auto"`. |
| `ipv4_dns` | string | optional | Array of IPv4 DNS such as `["192.168.0.254", "8.8.8.8"]`. Default: `[]`. |
| `ipv4_gateway` | string | optional | IPv4 gateway. Default: `""`. |
| `ipv4_route_metric` | integer | optional | IPv4 route metric. Lower values are preferred. Default: `0` which defaults to `100` for wired networks and `600` for wireless network. |
| `priority` | integer | optional | Priority to choose the network with. Values between -999 and 999 with higher values taking precedence. Default: `0`. |
| `psk` | string | optional | The network pass key. Default: `""`. |
| `ssid` | string | optional | The WiFi network's SSID. Only needed for WiFi networks. Default: `""`. |

> | `type` | string | optional | The type of the network. Options: `"wifi"`, `"wired"`. Default: `""`. |

Author: ```suggestion
| `type` | string | optional | The type of the network. Required if a network is provided. Options: `"wifi"`, `"wired"`. |
```

{{% /tab %}}
{{< /tabs >}}

## `version_control`: Version management for `viam-agent` and `viam-server`

> By default, when a new version of `viam-server` becomes available, `viam-agent` will restart and upgrade `viam-server` immediately.

Reviewer: I believe this is not accurate. agent will download the new version when available but it will not be used until agent restarts. This is why we introduce maintenance windows so the agent knows when is safe to do it automatically.

Reviewer: line 227 is accurate

Reviewer: Correct, either restarts viam-server or power cycles

| `hotspot_password` | string | false | The Wifi password for the provisioning hotspot. Default: `"viamsetup"`. |
| `hotspot_prefix` | string | false | `viam-agent` will prepend this to the hostname of the device and use the resulting string for the provisioning hotspot SSID. Default: `"viam-setup"`. |
| `manufacturer` | string | false | Purely informative. May be displayed on captive portal or provisioning app. Default: `"viam"`. |
| `model` | string | false | Purely informative. May be displayed on captive portal or provisioning app. Default: `"custom"`. |
| `offline_before_starting_hotspot_minutes` | integer | false | Will only enter provisioning mode (hotspot) after being disconnected longer than this time. Useful on flaky connections, or when part of a system where the device may start quickly, but the wifi/router may take longer to be available. Default: `2` (2 minutes). |
| `retry_connection_timeout_minutes` | integer | Optional | Provisioning mode will exit after this time, to allow other unmanaged (for example wired) or manually configured connections to be tried. Provisioning mode will restart if the connection/online status doesn't change. Default: `10` (10 minutes). |

> | `turn_on_hotspot_if_wifi_has_no_internet` | boolean | false | By default, the device will only attempt to connect to a single wifi network (the one with the highest priority), provided during initial provisioning/setup using the provisioning mobile app or captive web portal. Wifi connection alone is enough to consider the device as "online" even if the global internet is not reachable. If the primary network configured during provisioning cannot be connected to and `turn_on_hotspot_if_wifi_has_no_internet` is enabled, the device will attempt connections to all configured networks in `networks`, and only consider the device online if the internet is reachable. Default: `false`. |

Reviewer 2: ```suggestion
| `turn_on_hotspot_if_wifi_has_no_internet` | boolean | false | By default, the device will only attempt to connect to a single WiFi network (the one with the highest priority), provided during initial provisioning/setup using the provisioning mobile app or captive web portal. WiFi connection alone is enough to consider the device as "online" even if the global internet is not reachable. If the primary network configured during provisioning cannot be connected to and `turn_on_hotspot_if_wifi_has_no_internet` is enabled, the device will attempt connections to all configured networks in `networks`, and only consider the device online if the internet is reachable. Default: `false`. |
```

It's primarily useful for a machine that moves between different networks, so the machine can automatically connect when moved between locations.

<!-- prettier-ignore -->
| Name       | Type | Required? | Description |
| ---------- | ---- | --------- | ----------- |
| `interface` | string | optional | Name of interface, for example: `"wlan0"`, `"eth0"`, `"enp14s0"`. Default: `""`. |
| `ipv4_address` | string | optional | IPv4 address in CIDR format, for example: `"192.168.0.1/24"`. Default: `"auto"`. |
| `ipv4_dns` | string | optional | Array of IPv4 DNS such as `["192.168.0.254", "8.8.8.8"]`. Default: `[]`. |
| `ipv4_gateway` | string | optional | IPv4 gateway. Default: `""`. |
| `ipv4_route_metric` | integer | optional | IPv4 route metric. Lower values are preferred. Default: `0` which defaults to `100` for wired networks and `600` for wireless network. |
| `priority` | integer | optional | Priority to choose the network with. Values between -999 and 999 with higher values taking precedence. Default: `0`. |

> | `psk` | string | optional | The network pass key. Default: `""`. |

Reviewer 2: ```suggestion
| `psk` | string | optional | The network passkey. Default: `""`. |
```
I think?

{{< alert title="Important" color="note" >}}
You must enable `turn_on_hotspot_if_wifi_has_no_internet` in the [`agent-provisioning` configuration](#configuration) to allow the machine to connect to the specified networks.
Note that if you are using the Viam app to add networks to a machine's configuration, the machine will need to be connected to the internet to retrieve the configuration information containing the network credentials before it can use them.
{{< /alert >}}

If `turn_on_hotspot_if_wifi_has_no_internet` is enabled, `agent-provisioning` will try to connect to each specified network in order of `priority` from highest to lowest.
If the highest-priority network is not available or the machine can connect but internet is not available, `viam-agent` will then attempt to connect to the next-highest network, and so on until all configured networks have been tried.

## `system-configuration`

<!-- prettier-ignore -->
| Name       | Type | Required? | Description |
| ---------- | ---- | --------- | ----------- |
| `logging_journald_runtime_max_use_megabytes` | integer | optional |Set the temporary space limit for logs. `-1` to disable. Default: `512` (512 MB). |

> | `logging_journald_system_max_use_megabytes` | integer | optional | Sets the maximum disk space `journald` will user for persistent log storage. `-1` to disable. Default: `512` (512 MB). |

Reviewer 2: ```suggestion
| `logging_journald_system_max_use_megabytes` | integer | optional | Sets the maximum disk space `journald` will use for persistent log storage. `-1` to disable. Default: `512` (512 MB). |
```
